drivers/sensor/ti_hdc/ti_hdc.c

```diff
@@ -37,17 +39,22 @@ static int ti_hdc_sample_fetch(struct device *dev, enum sensor_channel chan)
 
 	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL);
 
-	gpio_pin_enable_callback(drv_data->gpio,
-				 DT_TI_HDC_0_DRDY_GPIOS_PIN);
+#if defined(DT_TI_HDC_0_DRDY_GPIOS_CONTROLLER)
+	gpio_pin_enable_callback(drv_data->gpio, DT_TI_HDC_0_DRDY_GPIOS_PIN);
+#endif
 
 	buf[0] = TI_HDC_REG_TEMP;
-	if (i2c_write(drv_data->i2c, buf, 1,
-		      DT_TI_HDC_0_BASE_ADDRESS) < 0) {
+	if (i2c_write(drv_data->i2c, buf, 1, DT_TI_HDC_0_BASE_ADDRESS) < 0) {
```

Comment:
This change seems unrelated to the commit message. Please make whitespace and formatting changes in a separate commit.